### PR TITLE
perlapi: Combine is_utf8_valid_partial_char, with _flags form

### PR DESCRIPTION
**Pull request by khwilliamson**

### CODE REVIEW

File: inline.h [hunk 1]
```diff
@@ -3111,32 +3115,24 @@ function is used to verify that the final bytes in the current buffer are in
...
 
+C<is_utf8_valid_partial_char> behaves identically to
+C<is_utf8_valid_partial_char_flags> when the latter is called with a zero
+C<flags> parameter.  This parameter is used to restrict the classes of code
+points that are considered to be valid.  When zero, Perl's extended UTF-8 is
+used.  Otherwise C<flags> can be any combination of the C<UTF8_DISALLOW_I<foo>>
+flags accepted by C<L</utf8n_to_uvchr>>.  If there is any sequence of bytes
+that can complete the input partial character in such a way that a
+non-prohibited character is formed, the function returns TRUE; otherwise FALSE.
+Non character code points cannot be determined based on partial character
```
Comment (jkeenan) [inline inline.h]: suggest: `s/Non character/Non-character/`

**Review by jkeenan — COMMENTED** [review]
Output from `perldoc pod/perlapi.pod` once branch is built LGTM.  Since content has been reworded, should get additional review.

File: inline.h [hunk 2]
```diff
@@ -3090,17 +3090,21 @@ Perl_isUTF8_CHAR_flags(const U8 * const s0, const U8 * const e, const U32 flags)
...
+further than S<C<e - 1>> is the UTF-8 encoding for one or more code points.
+That is, 0 is returned if C<s> points to at least one entire UTF-8 encoded
+character.
 
-In other words this returns TRUE if C<s> points to a partial UTF-8-encoded code
-point.
+Otherwise, they return 1 if there exists at least one non-empty sequence of
+bytes that when appended to sequence C<s>, starting at position C<e> causes the
+entire sequence to be the well-formed UTF-8 of some code point; otherwise
+returns 0.
```
Comment (jkeenan) [inline inline.h]: Sentence above contains two instances of `otherwise`.  The part after the semicolon can probably be cut.

File: inline.h (hunk 1 above)
Comment (khwilliamson) [reply inline.h]: done

File: inline.h (hunk 2 above)
Comment (khwilliamson) [reply inline.h]: done

File: inline.h [hunk 3]
```diff
@@ -3090,17 +3090,20 @@ Perl_isUTF8_CHAR_flags(const U8 * const s0, const U8 * const e, const U32 flags)
...
+character.
 
-In other words this returns TRUE if C<s> points to a partial UTF-8-encoded code
-point.
+Otherwise, they return 1 if there exists at least one non-empty sequence of
+bytes that when appended to sequence C<s>, starting at position C<e> causes the
+entire sequence to be the well-formed UTF-8 of some code point
+
+In other words they return TRUE if C<s> points to an incomplete UTF-8-encoded
+code point; FALSE otherwise.
```
Comment (tonycoz) [inline inline.h]: This was a problem in the original, but this is all inconsistent about returning 1 / 0 or TRUE / FALSE.

Comment (khwilliamson) [reply inline.h]: I also changed the code to return FALSE instead of 0  for the instance it did that.

**Review by tonycoz — APPROVED** [review]

**Pull request merged**